src/main/scala/Syntax.scala

```diff
@@ -0,0 +1,133 @@
+package fuselang
+
+import scala.util.parsing.input.Positional
+
+object Syntax {
+
+  import Errors._
+  import scala.math.{max,log10,ceil}
+
+  case class Id(v: String) extends Positional {
+    var typ: Option[Type] = None;
```

Comment:
It’s helpful to have types associated with identifiers for the emitter code to be kept simple.

Even without types Id isn’t inert because it carries positional information.